compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/PipelineTranslator.java

```diff
@@ -443,6 +442,12 @@ private static AbstractDoFnTransform createDoFnTransform(final PipelineTranslati
         Iterables.getOnlyElement(TransformInputs.nonAdditionalInputs(pTransform));
 
       final HasDisplayData displayData = (builder) -> builder.add(DisplayData.item("name", beamNode.getFullName()));
+      final DoFnSchemaInformation doFnSchemaInformation;
+      if (ctx.getCurrentTransform() != null) {
+        doFnSchemaInformation = ParDoTranslation.getSchemaInformation(ctx.getCurrentTransform());
+      } else {
+        doFnSchemaInformation = DoFnSchemaInformation.create();
```

Comment:
It's not straightforward to me why it creates a new schema information when the current transform is null.

Reply:
Handled.